src/sentry/snuba/metrics_layer/query.py

```diff
@@ -549,24 +559,26 @@ def fetch_metric_tag_keys(
     Fetches the tag keys for a given metric MRI. This will reverse
     resolve all the tag keys into strings.
     """
-    return _query_meta_table(org_id, project_ids, use_case_id, mri, app_id)
+    return _query_meta_table(org_id, project_ids, [use_case_id], mri, app_id)[use_case_id]
 
 
 def _query_meta_table(
     org_id: int,
     project_ids: list[int],
-    use_case_id: UseCaseID,
+    use_case_ids: Sequence[UseCaseID],
     mri: str | None = None,
     app_id: str = "",
-) -> dict[int, list[str]]:
+) -> dict[UseCaseID, dict[int, list[str]]]:
     """
     Helper function for querying the meta table. This will query across all four metric types, and resolve all the resulting
     values. If an MRI is provided, it is assumed that this function should find unique tag keys for that MRI.
     """
+    if len(use_case_ids) == 0:
+        raise InvalidParams("You must supply at least one use case id")
 
     if mri:
         column_name = "tag_key"
-        metric_id = resolve_weak(use_case_id, org_id, mri)
+        metric_id = resolve_weak(use_case_ids[0], org_id, mri)
```

Comment:
How does this work if the MRI has a different use case from `use_case_ids[0]`?

Reply:
It works in exactly the same way as your implementation, it will fail resolution.

Reply:
I see. This function has been overloaded and the correct call parameters are no longer as clear. I think there should be a check here that if an `mri` is provided, then exactly one `use_case_id` should be provided. Then it is clear what the intended use of this function is.